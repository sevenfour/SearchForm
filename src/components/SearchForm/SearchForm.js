import React, { useState } from 'react';
import PropTypes from 'prop-types';

import clsx from 'clsx';

import styles from './SearchForm.module.css';

const SearchForm = props => {

  const {
    className,
    error,
    onSubmit = () => {}
  } = props;

  const [searchQuery, setSearchQuery] = useState('');

  function handleChange(e) {
    setSearchQuery(e?.target?.value);
  }

  return (
    <form
      className={clsx(className, styles.form)}
      onSubmit={e => {
        e.preventDefault();
        onSubmit(searchQuery);
      }}
    >
      <div
        className={styles.formWrapper}
        role="search"
      >
        <label
          className={styles.label}
          htmlFor="search"
        >
          Search gear on MEC
          {
            error &&
              <p
                className={styles.error}
                role="alert"
              >
                {error}
              </p>
          }
        </label>
        <input
          aria-invalid={Boolean(error)}
          autoCapitalize="off"
          autoComplete="off"
          autoCorrect="on"
          className={styles.input}
          id="search"
          onChange={handleChange}
          type="search"
          value={searchQuery}
        />
        <button
          className={styles.button}
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
  );
};

SearchForm.propTypes = {
  className: PropTypes.string,
  error: PropTypes.string,
  onSubmit: PropTypes.func
};

export default SearchForm;
