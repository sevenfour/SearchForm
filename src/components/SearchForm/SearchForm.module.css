
.form {
  font-size: 1em;
  --max-width: 300px;
}

.formWrapper {
  display: flex;
  flex-direction: column;
}

.label {
  display: block;
  margin-bottom: 1em;
}

.input {
  border: 1px solid var(--asphalt-grey);
  padding: 0.85em;
  max-width: var(--max-width);
  height: 3em;
  min-height: 40px;
  font-size: 0.88em;
}

.input[ aria-invalid = 'true' ] {
  border-color: var(--watermelon-red);
}

.error {
  box-sizing: border-box;
  margin: 16px 0 0;
  border-radius: 4px;
  background-color: var(--watermelon-red);
  padding: 0.6em;
  max-width: var(--max-width);
  color: var(--white);
  font-size: 0.84em;
}

.button {
  margin-top: 32px;
  border: none;
  border-radius: 0;
  background-color: var(--stone-grey);
  padding: .8rem;
  max-width: 200px;
  color: var(--black);
  font-family: sans-serif;
  font-size: 0.88em;
  font-weight: 400;
}

.button:hover,
.button:focus {
  background-color: var(--dust-grey);
  cursor: pointer;
}
