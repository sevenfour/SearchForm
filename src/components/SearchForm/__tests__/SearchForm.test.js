import React from 'react';
import { shallow } from 'enzyme';

import SearchForm from '..';

describe('<SearchForm /> component tests', () => {

  it('renders', () => {
    const form = shallow(
      <SearchForm />
    );

    expect(form).toMatchSnapshot();
  });

  it('should call onSubmit', () => {

    const onSubmitSpy = jest.fn();

    const wrapper = shallow(
      <SearchForm onSubmit={onSubmitSpy} />
    );

    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect(onSubmitSpy).toHaveBeenCalled();
  });

  it('should call onSubmit with query', () => {

    const query = 'super-duper bike';
    const onSubmitSpy = jest.fn();

    const wrapper = shallow(
      <SearchForm onSubmit={onSubmitSpy} />
    );

    const input = wrapper.find('input');

    input.simulate('change', { target: { value: query } });
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect(onSubmitSpy).toHaveBeenCalledWith(query);
  });

});
