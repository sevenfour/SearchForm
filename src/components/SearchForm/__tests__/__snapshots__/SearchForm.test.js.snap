// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> component tests renders 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="formWrapper"
    role="search"
  >
    <label
      className="label"
      htmlFor="search"
    >
      Search gear on MEC
    </label>
    <input
      aria-invalid={false}
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="on"
      className="input"
      id="search"
      onChange={[Function]}
      type="search"
      value=""
    />
    <button
      className="button"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
